app/frontend/src/pages/page_helpers.ts

```
@@ -24,37 +32,262 @@ export async function setupStore(
+
+export function getChatReducer<A>() {
+    return function chatReducer(state: ChatState<A>, action: ChatAction<A>): ChatState<A> {
+        switch (action.type) {
+            case "SET_ANSWERS":
+                return { ...state, answers: action.payload };
+            case "ADD_ANSWER":
+                return { ...state, answers: [...state.answers, action.payload] };
+            case "UPDATE_LAST_ANSWER":
+                if (state.answers.length === 0) return state;
+                const newAnswers = [...state.answers];
+                newAnswers[newAnswers.length - 1] = action.payload;
```

Biome (1.9.4)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Unsafe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)